src/app/admin/manageUser/EditUserForm.tsx

```diff
@@ -18,8 +18,16 @@ interface Props {
     onConfirm: (alertOptions: AlertOptions) => void;
 }
 
+function isValidUserRole(userRole: UserRole | "UNKNOWN"): userRole is UserRole {
+    return userRole !== "UNKNOWN";
+}
+
 const EditUserForm: React.FC<Props> = (props) => {
-    const [role, setRole] = useState(props.userToEdit.userRole);
+    const initialRole = isValidUserRole(props.userToEdit.userRole)
+        ? props.userToEdit.userRole
+        : "caller";
+
+    const [role, setRole] = useState(initialRole);
```

Comment:
Let's specify type here to make it difficult to break logic around here
```suggestion
    const initialRole: UserRole = isValidUserRole(props.userToEdit.userRole)
        ? props.userToEdit.userRole
        : "caller";

    const [role, setRole] = useState<UserRole>(initialRole);
```